src/engine/console.cpp

```diff
@@ -940,55 +937,17 @@ void writecompletions(stream *f)
     }
 }
 
-#ifdef __APPLE__
-extern bool mac_capslock();
-extern bool mac_numlock();
-#endif
-
-bool capslockon = false, numlockon = false;
-#if !defined(WIN32) && !defined(__APPLE__)
-#include <X11/XKBlib.h>
-#endif
+bool capslockon = false;
 bool capslocked()
 {
-    #ifdef WIN32
-    if(GetKeyState(VK_CAPITAL)) return true;
-    #elif defined(__APPLE__)
-    if(mac_capslock()) return true;
-    #else
-    Display *d = XOpenDisplay((char*)0);
-    if(d)
-    {
-        uint n = 0;
-        XkbGetIndicatorState(d, XkbUseCoreKbd, &n);
-        XCloseDisplay(d);
-        return (n&0x01)!=0;
-    }
-    #endif
-    return false;
+    if((SDL_GetModState() & KMOD_CAPS) == KMOD_CAPS)
```

Comment:
You could also just check whether the result is non-zero. But this works just fine as well.

Why not a bit easier, though?

```cpp
    return (SDL_GetModState() & KMOD_CAPS) == KMOD_CAPS;
```

The return value is `bool`, not `int` anyway...